packages/remix-dev/config.ts

```diff
@@ -342,21 +369,67 @@ export async function readConfig(
     }
   }
 
-  let customServerEntryPoint = appConfig.server;
-  let serverBuildTarget: ServerBuildTarget | undefined =
-    appConfig.serverBuildTarget;
-  let serverModuleFormat: ServerModuleFormat =
-    appConfig.serverModuleFormat || "cjs";
-  let serverPlatform: ServerPlatform = appConfig.serverPlatform || "node";
+  // TODO: default to esm for v2
+  let serverModuleFormat: ServerModuleFormat = "cjs";
+  let serverPlatform: ServerPlatform = "node";
+  let serverConditions = undefined;
+  let serverMainFields = undefined;
+  let serverBuildPath = "build/index.js";
+  let defaultPublicPath = "/build/";
+  let serverDependenciesToBundle: "all" | Array<string | RegExp> | undefined;
+
+  // Retain backwards compatibility for `serverBuildTarget`
   switch (appConfig.serverBuildTarget) {
+    case "arc":
+      serverBuildPath = "server/index.js";
+      defaultPublicPath = "/_static/build/";
+      break;
     case "cloudflare-pages":
+      serverBuildPath = "functions/[[path]].js";
     case "cloudflare-workers":
+      serverConditions = ["worker"];
+      serverModuleFormat = "esm";
+      serverPlatform = "neutral";
+      serverConditions = ["worker"];
+      serverMainFields = ["browser", "module", "main"];
+      serverDependenciesToBundle = "all";
+      break;
     case "deno":
       serverModuleFormat = "esm";
       serverPlatform = "neutral";
+      serverConditions = ["deno", "module", "main"];
+      serverMainFields = ["module", "main"];
+      serverDependenciesToBundle = "all";
+      break;
+    case "netlify":
+      serverBuildPath = ".netlify/functions-internal/server.js";
+      break;
+    case "vercel":
+      serverBuildPath = "api/index.js";
       break;
   }
 
+  // override with new config if provided
+  serverConditions = appConfig.serverConditions
+    ? appConfig.serverConditions
+    : serverConditions;
+  serverMainFields = appConfig.serverMainFields
+    ? appConfig.serverMainFields
+    : serverMainFields;
+  serverDependenciesToBundle =
+    (appConfig.serverDependenciesToBundle
+      ? appConfig.serverDependenciesToBundle
+      : serverDependenciesToBundle) || [];
+
+  serverBuildPath = path.resolve(rootDirectory, serverBuildPath);
```

Comment:
You're still missing some things that I extracted into `resolveServerBuildPath` here

```ts
// retain deprecated behavior for now
  if (appConfig.serverBuildDirectory) {
    serverBuildPath = path.join(appConfig.serverBuildDirectory, "index.js");
  }

  if (appConfig.serverBuildPath) {
    serverBuildPath = appConfig.serverBuildPath;
  }
```